chia/consensus/blockchain.py

```diff
@@ -372,46 +398,63 @@ async def _reconsider_peak(
-                    if fetched_full_block.is_transaction_block():
-                        assert fetched_full_block.foliage_transaction_block is not None
-                        added_rec = await self.coin_store.new_block(
-                            fetched_full_block.height,
-                            fetched_full_block.foliage_transaction_block.timestamp,
-                            fetched_full_block.get_included_reward_coins(),
-                            tx_additions,
-                            tx_removals,
+                        tx_removals, tx_additions, npc_res = await self.get_tx_removals_and_additions(
+                            fetched_full_block, None
                         )
-                        removed_rec: List[Optional[CoinRecord]] = [
-                            await self.coin_store.get_coin_record(name) for name in tx_removals
-                        ]
-
-                        # Set additions first, than removals in order to handle ephemeral coin state
-                        # Add in height order is also required
-                        record: Optional[CoinRecord]
-                        for record in added_rec:
-                            assert record
-                            lastest_coin_state[record.name] = record
-                        for record in removed_rec:
-                            assert record
-                            lastest_coin_state[record.name] = record
+
+                    assert fetched_full_block.foliage_transaction_block is not None
+                    added_rec = await self.coin_store.new_block(
+                        fetched_full_block.height,
+                        fetched_full_block.foliage_transaction_block.timestamp,
+                        fetched_full_block.get_included_reward_coins(),
+                        tx_additions,
+                        tx_removals,
+                    )
+                    removed_rec: List[Optional[CoinRecord]] = [
+                        await self.coin_store.get_coin_record(name) for name in tx_removals
+                    ]
+
+                    # Set additions first, than removals in order to handle ephemeral coin state
+                    # Add in height order is also required
+                    record: Optional[CoinRecord]
+                    for record in added_rec:
+                        assert record
+                        lastest_coin_state[record.name] = record
+                    for record in removed_rec:
+                        assert record
+                        lastest_coin_state[record.name] = record
+
+                    if npc_res is not None:
+                        hint_list = self.get_hint_list(npc_res)
+                        await self.hint_store.add_hints(hint_list)
+                        # There can be multiple coins for the same hint
+                        for coin_name, hint in hint_list:
+                            key = hint
```

Comment:
Can we remove the "key" variable?

Reply:
avoids hashing same thing multiple times